templates/contribute/pr-review.md

```diff
@@ -0,0 +1,723 @@
+  fun x ↦ match this x.property with | ⟨y, hy⟩ => ⟨y, Subtype.ext hy⟩
+```
+
+In this case, appealing to the lemma `Algebra.adjoin_le` was a tremendous
+simplification over using `Algebra.adjoin_induction'`.
+
+#### Are the definitions introduced the best way to formalize the concept (very difficult!)?
+
+This is incredibly hard to describe in general, but perhaps the simplest
+rule of thumb that can be given is this: the more definitions avoid 
+dependent types, the better.
+
+As an example, consider the definition of [Vector](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Data/Vector.html#Vector).
+Many introductory expositions of dependent type theory define `Vector`
+as an inductive type and give it as the canonical example of a dependent
+type. While the dependent typing is unavoidable, mathlib's definition 
+instead opts for a simple subtype of `List`, and the dependence on `ℕ`
+appears only in `List.length` equality proposition. This has the 
+advantage that we can easily pass out of the dependent type world by
+coercing to `List` and then life is much easier. In fact, most operations
+on `Vector` are precisely the corresponding operation on `List` combined
+with an equality proof on the lengths.
+
+### Library integration
+
+Many of the questions here are a bit nebulous and generally require a large amount of
+familiarity with the structure of mathlib, or at least significant prior expertise
+in formalization. Don't be discouraged if you find it difficult to address these
+questions when reviewing.
+
+#### Does it provide a sensible API?
+
+- Are attributes added appropriately (e.g., `@[simp]`, `@[ext]`, `@[gcongr]`, `@[aesop]`, etc.)?
+- Are rewrite lemmas provided to avoid the need to pass through equalities definitionally all the time?
+- Does a new type provide convenient constructors in common use cases?
+
+#### Is it general enough to support known future needs?
+
+This requires knowing what some of the future needs may be! Being active on Zulip
+can help make a reviewer aware of these needs. However, one may use a proxy for this question, 
+namely, "is there a more general version of this result in the literature which invokes pre-existing
+concepts in mathlib?" If there is, perhaps the existing PR should be generalized.
+
+#### Does it fit the design and collective vision of mathlib?
+
+Again, this is a vague question and requires knowing what the design and collective vision are!
+However, as some practical examples:
+
+- If a user is adding a new kind of morphism (not in the category theory library), they should 
+  more likely than not be defining a bundled morphism type, and probably an associated morphism
+  class using the `FunLike` API.
+- Similarly, if a contributor is adding a new subobject, they should probably be using bundled
+  subobjects and making use of the `SetLike` API.
+- Follow the advice of any existing [library note](we-need-a-link-to-these).
```

Comment:
Maybe add a link to the [relevant chapter of MIL](https://leanprover-community.github.io/mathematics_in_lean/C07_Hierarchies.html)?